023-connected-accounts/README.md

```diff
@@ -0,0 +1,303 @@
+   * This will automatically create a connected account
+6. Create an OAuth2 session without creating an account (backwards compatibility)
+
+A developer should be able to:
+
+1. List a user's connected accounts
+2. Delete a user's connected account
+
+### API Endpoints
+
+<!--
+List the new API routes or endpoints that we might need to add for supporting the new feature.
+Keep in mind to stay very strict to the API protocol and method, whether your new
+changes are for the REST, WebSocket or any other API protocol Appwrite supports.
+
+For example:
+
+**POST /v1/coffee ** - an endpoint for creating coffee.
+**DELETE /v1/coffee ** - an endpoint for deleting coffee.
+-->
+
+#### Account Service
+
+##### Create Connected Account
+
+The existing Create OAuth2 Session endpoints will be re-used to create a connected account
+
+**GET /v1/account/sessions/oauth2/{provider}**
+
+##### List Connected Accounts
+
+**GET /v1/account/connected**
+
+Params:
+
+* queries: it should be possible to filter the queries by provider
+
+Response:
+
+List of [Connected Accounts](#connected-account)
+
+##### Get a Connected Account
+
+**GET /v1/account/connected/{id}**
+
+Response:
+
+[Connected Account](#connected-account)
+
+##### Refresh Connected Account
+
+**PATCH /v1/account/connected/{id}**
+
+Response:
+
+[Connected Account](#connected-account)
+
+##### Delete Connected Account
+
+**DELETE /v1/account/connected/{id}**
+
+Response:
+
+None
+
+#### Users Service
+
+##### List Connected Accounts
+
+**GET /v1/users/{userId}/connected**
+
+Params:
+
+* queries: it should be possible to filter the queries by provider
+
+Response:
+
+List of [Connected Accounts](#connected-account)
+
+##### Get a Connected Account
+
+**GET /v1/users/{userId}/connected/{id}**
+
+Response:
+
+[Connected Account](#connected-account)
+
+##### Refresh Connected Account
+
+**PATCH /v1/users/{userId}/connected/{id}**
+
+Response:
+
+[Connected Account](#connected-account)
+
+##### Delete Connected Account
+
+**DELETE /v1/users/{userId}/connected/{id}**
+
+Response:
+
+None
+
+### Data Structure
+
+<!--
+What kind of changes or additions are required for the Appwrite base collections
+to support this feature. Explain which entities should be added or updated, what new attributes they
+need to have and why. Please think well about the naming conventions and how well they play with other
+Appwrite conventions. Try and stay as consistent with existing patterns as much as possible.
+-->
+
+#### Connected Account
+
+* provider: OAuth2 provider
+* userId: user id from the provider
+* status: connected, disconnected
+* accessToken: access token from the provider
+* accessTokenExpiry: the date when the access token expires in ISO 8601 format
+* providerRefreshToken: refresh token from the provider
+
+### Flow
+
+#### User
+
+##### Connecting to an OAuth2 provider
+
+1. User makes authenticated API call to `GET /v1/account/sessions/oauth2/{provider}`
```

Comment:
Is it safe to assume that the main difference here from the existing flow is that if a user is already logged in with Appwrite when creating an OAuth session the OAuth connection will be attached to their current account regardless of the email?

Reply:
Yes, during the /redirect.